### PR TITLE
Glossary cleanup

### PR DESCRIPTION
**Pull request by stevepiercy**
Refs: https://github.com/plone/documentation/issues/1415 and https://github.com/plone/volto/pull/4211

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -194,14 +212,21 @@ Express
...
 
-Server-Side Rendering (SSR)
-    When first loading any Plone page, users will get HTML markup that closely matches the final DOM structure of the React components used to render that page.
+SSR
+server-side rendering
+    When first loading any Plone page, SSR will send HTML markup that closely matches the final DOM structure of the React components used to render that page.
+
+    More generally, when a client sends a request to an SSR page, the HTML for the page is created on the server on each request.
+    SSR enables a developer to customize a website per request and per user.
+    In addition, SSR can improve performance and search engine optimization (SEO) for a website.
```
Comment (jensens) [inline docs/glossary.md]: I would write it like so:
```suggestion
    When an HTTP client (web browser) sends a request, the HTML markup for the page is created on the server.
    
    When loading a page in  Volto the first time, SSR will send HTML markup that closely matches the final DOM structure of the React components used to render that page. 
    After initial loading the {term}`hydration` mechanism is taking over and then rendering happens browser side.
    
    When loading a page in ClassicUI the full page is loaded on every request (rarely HTML-snippets are loaded, i.e. for overlays).

    SSR enables a developer to customize a website per request and per user.
    In addition, SSR can improve performance and search engine optimization (SEO) for a website.
```

File: docs/glossary.md [hunk 2]
```diff
@@ -194,14 +212,21 @@ Express
...
 
-Single Page Application (SPA)
+SPA
+single page application
     A type of JavaScript application that aims to provide a better user experience by avoiding unnecessary reloading of the browser page, instead using AJAX to load backend information.
```
Comment (jensens) [inline docs/glossary.md]: Should we define `AJAX` too? https://en.wikipedia.org/wiki/Ajax_(programming)

File: docs/glossary.md [hunk 3]
```diff
@@ -235,16 +260,17 @@ JSX
...
-    They allow customizing the behavior of the data flow inside the redux store.
+Redux
+Redux middleware
+    Custom wrappers for the [Redux](https://redux.js.org/) store dispatch methods.
+    They allow customizing the behavior of the data flow inside the Redux store.
 
-hooks (React)
-    Hooks are a React API that allow function components to use React features such as lifecycle methods, states, and so on.
+hooks
+    [Hooks](https://reactjs.org/docs/hooks-overview.html) are a React API that allow function components to use React features, such as lifecycle methods, states, and so on.
```
Comment (jensens) [inline docs/glossary.md]: There are hooks in Python too. Should be like 
- What are hooks for in prgramming
- Hooks in React
- Hooks in Python

**Review by jensens — CHANGES_REQUESTED** [review]

File: docs/glossary.md (hunk 1 above)
Comment (jensens) [reply docs/glossary.md]: Also, DOM should be defined as term then.

Comment (stevepiercy) [reply docs/glossary.md]: Rewritten in [5ace25b](https://github.com/plone/documentation/pull/1416/commits/5ace25b34ae9dd83453632a092a971d38098c701). Please let me know if this hits the mark.

File: docs/glossary.md (hunk 2 above)
Comment (stevepiercy) [reply docs/glossary.md]: Done in [f466123](https://github.com/plone/documentation/pull/1416/commits/f4661232f14259df434967d0fc224cec98f2104a)

File: docs/glossary.md (hunk 3 above)
Comment (stevepiercy) [reply docs/glossary.md]: Rewritten in [3a96dbb](https://github.com/plone/documentation/pull/1416/commits/3a96dbbc9a2ac4d7d0f7498f76dfa3c7cab20a5d)

Comment (stevepiercy) [reply docs/glossary.md]: I forgot to mention, there are no specific Python hooks discussed in the docs, just general programming hooks and React hooks.

File: docs/glossary.md (hunk 1 above)
Comment (stevepiercy) [reply docs/glossary.md]: > Also, DOM should be defined as term then.

Done in [c538e9c](https://github.com/plone/documentation/pull/1416/commits/c538e9cde2e2b7001b5f433ed3d63671120a7ff2)

**Review by jensens — APPROVED** [review]

**Pull request merged**